dockstore-integration-testing/src/test/java/io/dockstore/client/cli/CRUDClientIT.java

```diff
@@ -263,6 +263,60 @@ public void testUpdatingHostedTool() {
+    @Test
+    public void testUpdatingDefaultVersionHostedTool() {
+        ApiClient webClient = getWebClient();
+        ContainersApi containersApi = new ContainersApi(webClient);
+        HostedApi hostedApi = new HostedApi(webClient);
+
+        // Add a tool with a version
+        DockstoreTool hostedTool = hostedApi.createHostedTool("awesomeTool", "cwl", "quay.io", "coolNamespace");
+        SourceFile descriptorFile = new SourceFile();
+        descriptorFile.setContent("cwlVersion: v1.0\\nclass: CommandLineTool\\nbaseCommand: echo\\ninputs:\\nmessage:\\ntype: string\\ninputBinding:\\nposition: 1\\noutputs: []");
+        descriptorFile.setType(SourceFile.TypeEnum.DOCKSTORE_CWL);
+        descriptorFile.setPath("/Dockstore.cwl");
+        SourceFile dockerfile = new SourceFile();
+        dockerfile.setContent("FROM ubuntu:latest");
+        dockerfile.setType(SourceFile.TypeEnum.DOCKERFILE);
+        dockerfile.setPath("/Dockerfile");
+        DockstoreTool dockstoreTool = hostedApi.editHostedTool(hostedTool.getId(), Lists.newArrayList(descriptorFile, dockerfile));
+        Optional<Tag> first = dockstoreTool.getTags().stream().max(Comparator.comparingInt((Tag t) -> Integer.parseInt(t.getName())));
+        Assert.assertEquals("correct number of source files", 2, first.get().getSourceFiles().size());
+
+        // Update the default version of the tool
+        if (first.isPresent()) {
+            containersApi.updateDefaultVersion(hostedTool.getId(), first.get().getName());
+        }
+    }
+
+    /**
+     * Ensures that hosted workflows can have their default path updated
+     */
+    @Test
+    public void testUpdatingDefaultVersionHostedWorkflow() {
+        ApiClient webClient = getWebClient();
+        WorkflowsApi workflowsApi = new WorkflowsApi(getWebClient());
+        HostedApi hostedApi = new HostedApi(webClient);
+
+        // Add a workflow with a version
+        Workflow hostedWorkflow = hostedApi.createHostedWorkflow("awesomeTool", "cwl", null, null);
+        SourceFile file = new SourceFile();
+        file.setContent("cwlVersion: v1.0\\nclass: Workflow\\ninputs:\\ninp: File\\nex: string\\noutputs:\\nclassout:\\ntype: File\\noutputSource: compile/classfile\\nsteps:\\nuntar:\\nrun: tar-param.cwl\\nin:\\ntarfile: inp\\nextractfile: ex\\nout: [example_out]\\ncompile:\\nrun: arguments.cwl\\nin:\\nsrc: untar/example_out\\nout: [classfile]");
+        file.setType(SourceFile.TypeEnum.DOCKSTORE_CWL);
+        file.setPath("/Dockstore.cwl");
+        Workflow dockstoreWorkflow = hostedApi.editHostedWorkflow(hostedWorkflow.getId(), Lists.newArrayList(file));
+        Optional<WorkflowVersion> first = dockstoreWorkflow.getWorkflowVersions().stream().max(Comparator.comparingInt((WorkflowVersion t) -> Integer.parseInt(t.getName())));
+        Assert.assertEquals("correct number of source files", 1, first.get().getSourceFiles().size());
+
+        // Update the default version of the workflow
+        if (first.isPresent()) {
```

Comment:
same here, not best practice in normal code, but as a test its ok
(or do `assertTrue(first.isPresent())`)